### PR TITLE
IVDCM communication with mobile; CM-80; CM-81; CM-85; CM-92; CM-197; CM-198; CM-306

### PR DESCRIPTION
**Pull request by VladSemenyuk**

### CODE REVIEW

File: src/components/ivdcm_module/include/ivdcm_module/commands/on_internet_state_changed_notification.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_COMMANDS_ON_INTERNET_SATE_CHANGED_NOTIFICATION_H_
+#define SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_COMMANDS_ON_INTERNET_SATE_CHANGED_NOTIFICATION_H_
+
+#include "ivdcm_module/ivdcm.h"
```
Comment (DKlimenko) [inline src/components/ivdcm_module/include/ivdcm_module/commands/on_internet_state_changed_notification.h]: forward declaration

**Comment by DKlimenko** [comment]
reviewed

File: src/components/ivdcm_module/include/ivdcm_module/commands/send_ivdcm_data_request.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_SEND_IVDCM_DATA_REQUEST_H_
+#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_SEND_IVDCM_DATA_REQUEST_H_
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/commands/send_ivdcm_data_request.h]: Wrong name. You skipped commands directory.

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event_dispatcher.h [hunk 3]
```diff
@@ -39,12 +39,12 @@
...
 
-#include "can_cooperation/event_engine/event.h"
-#include "can_cooperation/event_engine/event_observer.h"
+#include "ivdcm_module/event_engine/event.h"
+#include "ivdcm_module/event_engine/event_observer.h"
 
 #include "interfaces/MOBILE_API.h"
 
-namespace ivdcm_module {
+namespace event_engine {
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event_dispatcher.h]: I don't understand why you change this name of this namespace.

File: src/components/ivdcm_module/include/ivdcm_module/ivdcm.h [hunk 4]
```diff
@@ -67,26 +68,64 @@ class Ivdcm
...
+   * @brief Sends message to mobile application
+   * @param msg message
+   */
+  void SendMessageToMobile(application_manager::MessagePtr msg);
+
+  // RequestController wrapped methods
+  void AddRequestToRequestController(const uint32_t& mobile_correlation_id,
+                                  request_controller::MobileRequestPtr request);
+  void DeleteRequestFromRequestController(const uint32_t& mobile_correlation_id);
+  void ResetTimer(const uint32_t& mobile_correlation_id);
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/ivdcm.h]: I think you don't need to use reference.

**Comment by bonewell** [comment]
Reviewed.

**Pull request merged**